File: contains_duplicate.kt
class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        val map = mutableMapOf<Int, Int>();

        for (i in nums) {
            if (map.containsKey(i)){
                return true
            } else {
                map.put(i, 1)
            }
        }
        return false
    }
}

File: contains_duplicate.cxx
#include <vector>
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> match;

        // O(n) loop over nums
        for (int num : nums) {
            if (match.count(num) > 0) {
                return true;
            } 
            match.insert(num);
        }
        return false;
    }
};